grub-core/lib/gpt.c

```diff
@@ -237,6 +249,64 @@ grub_gpt_header_check (struct grub_gpt_header *gpt,
 }
 
 static grub_err_t
+grub_gpt_check_primary (grub_gpt_t gpt)
```

Comment:
Everything looks fine, except I'm curious why you didn't just add logged variant of `grub_gpt_sector_to_addr` with the same signature just with the addition of a context string for "primary" vs. "secondary" to stick in the logs.  The current code seems a bit of unnecessarily copy-and-paste